### PR TITLE
Github Actions and fixes so they pass.

### PR DESCRIPTION
**Pull request by edwarnicke**
Not sure if you want to switch cloudtest from circleci to github actions... but this pr would do it.

### CODE REVIEW

File: pkg/commands/execution_context.go [hunk 1]
```diff
@@ -301,7 +301,10 @@ func (ctx *executionContext) performShutdown() {
 				go func() {
 					defer ctx.clusterWaitGroup.Done()
 					logrus.Infof("Closing cluster %v %v", group.config.Name, curInst.id)
-					ctx.destroyCluster(curInst, false, false)
+					// TODO - what should we be doing with this error?
+					if err := ctx.destroyCluster(curInst, false, false);err != nil {
```
Comment (edwarnicke) [inline pkg/commands/execution_context.go]: I really don't know what the right thing to do with the error here is, so logged it.  If that's the right thing, I will remove the TODO comment and repush, just let me know.

File: pkg/commands/execution_context.go [hunk 2]
```diff
@@ -1143,7 +1146,10 @@ func (ctx *executionContext) monitorCluster(context context.Context, ci *cluster
 		err := ci.instance.CheckIsAlive()
 		if err != nil {
 			logrus.Errorf("Failed to interact with %s: %v", ci.id, err)
-			ctx.destroyCluster(ci, true, false)
+			// TODO what should we do about this error?
```
Comment (edwarnicke) [inline pkg/commands/execution_context.go]: I really don't know what the right thing to do with the error here is, so logged it. If that's the right thing, I will remove the TODO comment and repush, just let me know.

File: pkg/providers/packet/packet_provider.go [hunk 3]
```diff
@@ -463,15 +463,16 @@ func (pi *packetInstance) GetRoot() string {
...
-func (pi *packetInstance) doDestroy(writer io.StringWriter, timeout time.Duration, err error) {
-	_, _ = writer.WriteString(fmt.Sprintf("Error during k8s API initialisation %v", err))
-	_, _ = writer.WriteString(fmt.Sprintf("Trying to destroy cluster"))
-	// In case we failed to start and create cluster utils.
-	err2 := pi.Destroy(timeout)
-	if err2 != nil {
-		_, _ = writer.WriteString(fmt.Sprintf("Error during destroy of cluster %v", err2))
-	}
-}
+// TODO - this function was unused.  Should we delete it or should it be being used?
```
Comment (edwarnicke) [inline pkg/providers/packet/packet_provider.go]: This function was unused.  If its really unused, we should remove it.  If it should be used, let me know where and how :)

File: pkg/providers/shell/shell_provider.go [hunk 4]
```diff
@@ -202,15 +201,16 @@ func (si *shellInstance) GetRoot() string {
...
-func (si *shellInstance) doDestroy(writer io.StringWriter, timeout time.Duration, err error) {
-	_, _ = writer.WriteString(fmt.Sprintf("Error during k8s API initialisation %v", err))
-	_, _ = writer.WriteString(fmt.Sprintf("Trying to destroy cluster"))
-	// In case we failed to start and create cluster utils.
-	err2 := si.Destroy(timeout)
-	if err2 != nil {
-		_, _ = writer.WriteString(fmt.Sprintf("Error during destroy of cluster %v", err2))
-	}
-}
+// TODO - this function was unused.  Should we delete it or should it be being used?
```
Comment (edwarnicke) [inline pkg/providers/shell/shell_provider.go]: This function was unused.  If its really unused, we should remove it.  If it should be used, let me know where and how :)

File: pkg/commands/execution_context.go (hunk 1 above)
Comment (haiodo) [reply pkg/commands/execution_context.go]: It is used only when we start cluster and if start and stop are failed, we stop retrying to create cluster for this provider. Error is logged inside, so we could have only `_ = ctx.destroyClusetr(...) ` here.

File: pkg/commands/execution_context.go (hunk 2 above)
Comment (haiodo) [reply pkg/commands/execution_context.go]: Error is logged inside, I will fix this with next PR.

File: pkg/providers/packet/packet_provider.go (hunk 3 above)
Comment (haiodo) [reply pkg/providers/packet/packet_provider.go]: Yes, it is really not used anymore.

File: pkg/providers/shell/shell_provider.go (hunk 4 above)
Comment (haiodo) [reply pkg/providers/shell/shell_provider.go]: Same here, not used :-) Thanks.

**Comment by haiodo** [comment]
@edwarnicke I would be really glad to switch to use Github actions.
Github actions script only leak support for checking for formatter:

```
   name: Check code formatting
          command: |
            make install-formatter
            make format
            git diff --exit-code || (echo 'Seems like your Go files are not properly formatted. Run `make format` in your branch and commit the changes.' && exit 1)
  
```

**Pull request merged**